src/components/Card.js

```diff
@@ -1,25 +1,43 @@
 class Card {
-  constructor(data, templateSelector, handleImageClick) {
+  constructor(
+    data,
+    templateSelector,
+    handleImageClick,
+    handleDeleteButton,
+    handleLike
+  ) {
     this._name = data.name;
     this._link = data.link;
     this._templateSelector = templateSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteButton = handleDeleteButton;
+    this._id = data._id;
+    this._handleLike = handleLike;
+    this._isLiked = false;
   }
 
   _setEventListeners() {
     this._likeButton = this._element.querySelector(".card__like-button");
     this._deleteButton = this._element.querySelector(".card__remove-button");
-    this._likeButton.addEventListener("click", () => this._handleLikeButton());
-    this._deleteButton.addEventListener("click", () =>
-      this._handleDeleteCard()
-    );
+    this._likeButton.addEventListener("click", () => {
+      if (this._likeButton.classList.contains("card__like-button_active")) {
+        this._handleLike(this.getId());
+        this._likeButton.classList.remove("card__like-button_active");
+      } else {
+        this._likeButton.classList.add("card__like-button_active");
+        this._handleLike(this.getId());
+      }
+    });
+    this._deleteButton.addEventListener("click", () => {
+      this._handleDeleteButton(this.getId(), this._element);
+    });
     this._cardImage.addEventListener("click", () => {
       this._handleImageClick(this._name, this._link);
     });
   }
 
```

Comment:
Needs correction: Don't forget to set the initial likes button state when initializing the card